grudge/models/advection.py

```diff
@@ -279,18 +279,17 @@ def to_quad(arg):
 # {{{ closed surface advection
 
 def v_dot_n_tpair(actx, dcoll, velocity, trace_dd):
-    from grudge.dof_desc import DTAG_BOUNDARY
+    from grudge.dof_desc import BoundaryDomainTag
     from grudge.trace_pair import TracePair
     from meshmode.discretization.connection import FACE_RESTR_INTERIOR
 
     normal = actx.thaw(dcoll.normal(trace_dd.with_discr_tag(None)))
     v_dot_n = velocity.dot(normal)
     i = op.project(dcoll, trace_dd.with_discr_tag(None), trace_dd, v_dot_n)
 
-    if trace_dd.domain_tag is FACE_RESTR_INTERIOR:
-        e = dcoll.opposite_face_connection()(i)
-    elif isinstance(trace_dd.domain_tag, DTAG_BOUNDARY):
-        e = dcoll.distributed_boundary_swap_connection(trace_dd)(i)
+    assert isinstance(trace_dd.domain_tag, BoundaryDomainTag)
+    if trace_dd.domain_tag.tag is FACE_RESTR_INTERIOR:
+        e = dcoll.opposite_face_connection(trace_dd.domain_tag)(i)
```

Comment:
This was originally removed in #236, I guess in part because `distributed_boundary_swap_connection` was also removed. I didn't remove that method in this PR, so I restored the call here. I'm a little bit confused about what's actually going on in this code though... I thought the boundary swap connection was remote-to-local, not local-to-remote as it appears to be used here?

Reply:
Sorry, I shouldn't have said local-to-remote -- that wouldn't make sense either. What I meant is that it's applying a connection that goes remote-to-local to something that's already on the local boundary discr (`i`). That seems at best redundant (if the discrs are the same) and at worst wrong (if they're different), right?